kraken-build/src/kraken/std/cargo/tasks/cargo_publish_task.py

```diff
@@ -53,3 +65,48 @@ def make_safe(self, args: list[str], env: dict[str, str]) -> None:
     def __init__(self, name: str, project: Project) -> None:
         super().__init__(name, project)
         self._base_command = ["cargo", "publish"]
+
+    def _get_updated_cargo_toml(self, version: str) -> str:
+        from kraken.std.cargo.manifest import CargoManifest
+
+        manifest = CargoManifest.read(self.cargo_toml_file.get())
+        if manifest.package is None:
+            return manifest.to_toml_string()
+
+        # Cargo does not play nicely with semver metadata (ie. 1.0.1-dev3+abc123)
+        # We replace that to 1.0.1-dev3abc123
+        fixed_version_string = version.replace("+", "")
+        manifest.package.version = fixed_version_string
+        if manifest.workspace and manifest.workspace.package:
+            manifest.workspace.package.version = version
+
+        if self.registry.is_filled():
+            CargoProject.get_or_create(self.project)
+            registry = self.registry.get()
+            if manifest.dependencies:
+                self._push_version_to_path_deps(fixed_version_string, manifest.dependencies.data, registry.alias)
+            if manifest.build_dependencies:
+                self._push_version_to_path_deps(fixed_version_string, manifest.build_dependencies.data, registry.alias)
+        return manifest.to_toml_string()
+
+    def _push_version_to_path_deps(
+        self, version_string: str, dependencies: dict[str, Any], registry_alias: str
+    ) -> None:
+        """For each dependency in the given dependencies, if the dependency is a `path` dependency, injects the current
+        version and registry (required for publishing - path dependencies cannot be published alone)."""
+        for dep_name in dependencies:
+            dependency = dependencies[dep_name]
+            if isinstance(dependency, dict):
+                if "path" in dependency:
+                    dependency["version"] = f"={version_string}"
+                    dependency["registry"] = registry_alias
+
+    def execute(self) -> TaskStatus:
+        with contextlib.ExitStack() as stack:
+            if (version := self.version.get()) is not None:
+                content = self._get_updated_cargo_toml(version)
+                fp = stack.enter_context(atomic_file_swap(self.cargo_toml_file.get(), "w", always_revert=True))
+                fp.write(content)
+                fp.close()
+                result = super().execute()          
```

Comment:
This has one too many indent. We want to run super.execute regardless of whether a version is set to bump or not.

Reply:
Alright, that's what i thought. But then i won't get around writing `super().execute()` twice i think

Reply:
Or this?
```
result = TaskStatus.pending()
with contextlib.ExitStack() as stack:
    if (version := self.version.get()) is not None:
        content = self._get_updated_cargo_toml(version)
        fp = stack.enter_context(atomic_file_swap(self.cargo_toml_file.get(), "w", always_revert=True))
        fp.write(content)
        fp.close()
    result = super().execute()

return result
```

Reply:
Ok, now i understand. Didn't know this was possible without first initializing the result variable.